src/components/Contacts/ContactDetails/ContactDetailsTab/Mailing/EditContactAddressModal/EditContactAddressModal.tsx

```diff
@@ -0,0 +1,276 @@
+import React, { ReactElement } from 'react';
+import { Formik } from 'formik';
+import { useTranslation } from 'react-i18next';
+import * as yup from 'yup';
+import { useSnackbar } from 'notistack';
+import {
+  Box,
+  Button,
+  Checkbox,
+  CircularProgress,
+  DialogActions,
+  DialogContent,
+  FormControl,
+  FormControlLabel,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Select,
+  styled,
+  TextField,
+} from '@material-ui/core';
+import { ContactMailingFragment } from '../ContactMailing.generated';
+import { AddressUpdateInput } from '../../../../../../../graphql/types.generated';
+import Modal from '../../../../../common/Modal/Modal';
+import { useUpdateContactAddressMutation } from './EditContactAddress.generated';
+
+const ContactEditContainer = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  width: '100%',
+  flexDirection: 'column',
+  margin: theme.spacing(4, 0),
+}));
+
+const ContactInputWrapper = styled(Box)(({ theme }) => ({
+  position: 'relative',
+  padding: theme.spacing(0, 6),
+  margin: theme.spacing(2, 0),
+}));
+
+const LoadingIndicator = styled(CircularProgress)(({ theme }) => ({
+  margin: theme.spacing(0, 1, 0, 0),
+}));
+
+interface EditContactAddressModalProps {
+  accountListId: string;
+  address: ContactMailingFragment['addresses']['nodes'][0];
+  handleClose: () => void;
+}
+
+enum AddressLocationEnum {
+  Home = 'Home',
+  Business = 'Business',
+  Mailing = 'Mailing',
+  Seasonal = 'Seasonal',
+  Other = 'Other',
+  Temporary = 'Temporary',
+  RepAddress = 'Rep Address',
+}
+
+export const EditContactAddressModal: React.FC<EditContactAddressModalProps> = ({
+  accountListId,
+  address,
+  handleClose,
+}): ReactElement<EditContactAddressModalProps> => {
+  const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const [
+    updateContactAddress,
+    { loading: updating },
+  ] = useUpdateContactAddressMutation();
+
+  // TODO Let Spencer know of contactId bug
```

Comment:
A small bug I found here. The input type allows for a ```contactId``` so I just included it initially. However, it throws an API error when I include it.
<img width="797" alt="Screen Shot 2021-07-26 at 3 06 42 PM" src="https://user-images.githubusercontent.com/39680460/127052787-96964322-332e-4e21-94d5-49dda31ea04b.png">

Reply:
Should we make a ticket? Is it blocking you? Do you need to do something client side after? I just don't really want this comment merged since it isn't actionable.

Reply:
No this doesn't seem to be blocking me. I can make a ticket for Spencer. I think the only thing for the client to do after is to just remove the ```Omit<AddressUpdateInput, 'contactId'>```. I can make a comment that better reflects that. Should I also include a link to the ticket in the comment?

Reply:
Sure, I think it's always helpful to have  a way of knowing when a comment has been resolved or how to resolve it.

Reply:
Okay I went ahead and made [the ticket](https://jira.cru.org/browse/MPDX-7090) and updated the comment.